src/Compilers/CSharp/Portable/Binder/Binder_Operators.cs

```diff
@@ -627,6 +628,30 @@ private void ReportAssignmentOperatorError(AssignmentExpressionSyntax node, Diag
 
         private static void ReportBinaryOperatorError(ExpressionSyntax node, DiagnosticBag diagnostics, SyntaxToken operatorToken, BoundExpression left, BoundExpression right, LookupResultKind resultKind)
         {
+            if (resultKind == LookupResultKind.Empty && operatorToken.Kind() == SyntaxKind.DotDotToken)
+            {
+                var leftSpecialType = left.Type.SpecialType;
+                var rightSpecialType = right.Type.SpecialType;
+                if (leftSpecialType.IsIntegralType() && rightSpecialType.IsIntegralType())
+                {
+                    // User meant to use the built-in range type, but it was missing
+                    if (leftSpecialType == SpecialType.System_Int64 || leftSpecialType == SpecialType.System_UInt64 ||
+                        rightSpecialType == SpecialType.System_Int64 || rightSpecialType == SpecialType.System_UInt64)
+                    {
+                        Error(diagnostics, ErrorCode.ERR_RangeNotFound, node, WellKnownTypes.GetMetadataName(WellKnownType.System_LongRange));
+                    }
+                    else
+                    {
+                        Error(diagnostics, ErrorCode.ERR_RangeNotFound, node, WellKnownTypes.GetMetadataName(WellKnownType.System_Range));
+                    }
+                    return;
+                }
+                else
+                {
+                    // User meant to use a custom range operator: fall through to ERR_BadBinaryOps
```

Comment:
Please add a PROTOTYPE comment to track this.

Reply:
Hrm. This isn't a prototype comment, it's intended behavior. I'll reword the comment. Here's the situation:

1) Okay, we're in a state where no applicable Range constructors were found.
1) Were both arguments `int`s? (or `long`s, `short`s, etc) If so, `System.*Range` is probably missing.
    1) Was either argument a `long` or `ulong`? If so, it's probably `System.LongRange` that's missing. Report an error on that
    1) Otherwise, it's probably `System.Range` that's missing. Report an error on that.
1) Otherwise, this Range-failure is completely unrelated to the built-in ranges (because both sides are not both `int`-ish)
    1) We already have standard error reporting for "uh oh, this operator is missing! panic!!"
        1) This is the "fall through" comment: fall through to the logic that throws `ERR_BadBinaryOps`, which is just beneath this in this method.

Reply:
My bad, just didn't read it correctly.